lib/spack/spack/ci/common.py

```diff
@@ -40,16 +44,52 @@
 _urlopen = web_util.urlopen
 
 
-def copy_files_to_artifacts(src, artifacts_dir):
+def copy_gzipped(glob_or_path, dest):
+    """Copy all of the files in the source glob/path to the destination.
+
+    Parameters:
+        glob_or_path: path to file to test
+        dest: destination path to copy to
+    """
+
+    files = glob.glob(glob_or_path)
+    if not files:
+        raise OSError("No such file or directory: '{0}'".format(glob_or_path))
+    if len(files) > 1 and not os.path.isdir(dest):
+        raise ValueError(
+            "'{0}' matches multiple files but '{1}' is not a directory".format(glob_or_path, dest)
+        )
+
+    def is_gzipped(path):
+        with open(path, "rb") as fd:
+            return compression.GZipFileType().matches_magic(fd)
+
+    for src in files:
+        if is_gzipped(src):
+            fs.copy(src, dest)
+        else:
+            # Compress and copy in one step
+            src_name = os.path.basename(src)
+            zipped = os.path.join(dest, f"{src_name}.gz")
+            with open(src, "rb") as fin, gzip.open(zipped, "wb") as fout:
+                shutil.copyfileobj(fin, fout)
+
+
+def copy_files_to_artifacts(src, artifacts_dir, *, compress_artifacts=False):
```

Comment:
Instead of adding types to the docstring, you can add types to the function signature so mypy can actually enforce them:
```suggestion
def copy_files_to_artifacts(src: str, artifacts_dir: str, *, compress_artifacts: bool = False) -> None:
```